app/controllers/user_credentials_controller.rb

```diff
@@ -14,6 +14,19 @@ def index
     end
   end
 
+  def import
+    file = params[:file]
+    content = JSON.parse(file.read)
+    content.each do |hash|
+      @user_credential = current_user.user_credentials.build(hash.except!("id")).save!
```

Comment:
We have a [uniqueness constraint](https://github.com/Jngai/huginn/blob/uploadcredentialsfeature/app/models/user_credential.rb#L12) on the credential name. So we need need to handle the case where the `save` operation fails and avoid throwing an exception. You could add the imported user credentials to the association like this:
```
current_user.user_credentials << current_user.user_credentials.build(hash.except!("id"))
```
then use `if current_user.save` in the `respond_to` block and handle the failed save there.